homeassistant/components/refoss/__init__.py

```diff
@@ -0,0 +1,95 @@
+"""Refoss devices platform loader"""
+from __future__ import annotations
+
+from typing import Collection
+from datetime import timedelta
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_MAC
+from homeassistant.core import HomeAssistant
+from refoss_ha.http_device import HttpDeviceInfo
+from refoss_ha.const import (
+    DOMAIN,
+    LOGGER,
+    DEVICE_LIST_COORDINATOR,
+    SOCKET_DISCOVER_UPDATE_INTERVAL,
+    PLATFORMS,
+)
+from .coordinator import RefossCoordinator
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    if not config_entry.data.get(CONF_MAC):
+        LOGGER.warning(
+            (
+                "The config entry %s probably comes from a custom integration, please"
+                " remove it if you want to use core refoss integration"
+            ),
+            config_entry.title,
+        )
+        return False
+    hass.data[DOMAIN] = {}
```

Comment:
This will reset the domain data every time a new device is added

Reply:
This will only run once, and the addition of new devices will be automatically discovered here  ?:
def _poll_discovered_device(*args, **kwargs):
        discovered_devices = refoss_coordinator.data

        known_devices = refoss_coordinator.find_devices()

        if _check_new_discovered_device(known_devices, discovered_devices.values()):
            hass.create_task(
                refoss_coordinator.async_device_discovery(
                    cached_http_device_list=discovered_devices.values()
                )
            )
    refoss_coordinator.async_add_listener(_poll_discovered_device)

Reply:
As soon as you add another config entry it will run again and the second one will overwrite the data for the first one

Reply:
When adding again, In config_flow.py, if entry exists,  execute  function:_abort_if_unique_id_configured(),  Just quit.

Reply:
hass is a singleton. Every hass.data is the same. When you add the second device it will set override the previous data when an empty dict